File: StringToInteger/StringToInteger.py
class Solution(object):
    def myAtoi(self, str):
        """
        :type str: str
        :rtype: int
        """
        INT_MAX = 2147483647
        INT_MIN = -2147483648
        res = 0
        flag = 1
        i = 0
        str += 'N'
        while i < len(str):
            while str[i] == " ":
                i += 1
            if str[i] == '-':
                flag = -1
                i += 1
            elif str[i] == '+':
                i += 1
            while str[i] >= '0' and str[i] <= '9':
                res = res*10 + int(str[i])
                i += 1
                if res*flag > INT_MAX:
                    return INT_MAX
                elif res*flag < INT_MIN:
                    return INT_MIN
            return res*flag
            

File: StringToInteger/StringToInteger.cpp
#include <string>

using std::string;

class Solution {
public:
	int myAtoi(string str) {
		long result = 0;
		int indicator = 1;
		int i = 0;
		while (str[i] == ' ')
			i++;
		if (str[i] == '-' || str[i] == '+')
			indicator = (str[i++] == '-') ? -1 : 1;
		while (i < str.size() && ('0' <= str[i] && str[i] <= '9')) {
			result = result * 10 + (str[i++] - '0');
			if (result * indicator >= INT_MAX) return INT_MAX;
			if (result * indicator <= INT_MIN) return INT_MIN;
		}
		return result * indicator;
	}
};